### ISSUE TITLE
Two preprocessing/encoder defects affecting the trained model

### ISSUE THREAD

**Issue by azadeh-gh** [NOAA-EMC/ocelot#119]
Both were found while preparing the FSOI observation-impact experiments, but neither is an
FSOI bug — they affect what the model is trained on and what it can perceive, so they matter
for any training run. They are independent and can be reviewed separately.

Fix branch: `fix/obs-qc-and-channel-mask`

---

### What happens

The prepbufr quality marker for `dewPointTemperature` changes character with height:

| Pressure band | Flags present | Share flagged 2 (good) |
|---|---|---|
| ≤ 275 hPa | 9 (not checked), few 14 | **0.0%** |
| exactly 300 hPa | 2 (good), 15 (bad), few 14 | 77.8% |
| 275–350 hPa | 2, 9, 15, few 13/14 | 58.6% |
| 350–450 hPa | 2, few 3/13/14/15 | 96.8% |
| 900–1100 hPa | 2, few 3/13/14 | dominant |

Flag 9 means the observation was **not checked and not used by the analysis**. Operational
practice does not assimilate radiosonde moisture above 300 hPa, because the humidity sensor
loses skill at stratospheric temperatures. Below and at 300 hPa the data are checked: values
are either accepted (2) or rejected (15).

These are not fill values — at ≤ 30 hPa the median dewpoint is −78.9 °C with a median T−Td
spread of 33 °C, and `Td == T` essentially never occurs. They are real numbers that nothing
has validated.

### Why they survive preprocessing

In `configs/observation_config.yaml`, radiosonde `dewPointTemperature` QC was a range of
`[-120, 50]` °C plus `reject: [13, 14, 15]`. Flag 9 is not in that list, and
`qc_strict_flags` is not set for radiosonde, so unchecked values pass. The target mask is
then built as `~isnan(...)`, so every one of them becomes a scored target.

### Impact

- **Training:** the loss fits unchecked stratospheric humidity. Radiosonde is one of four
  target channels, so above 300 hPa a quarter of the radiosonde target signal is unvetted.
- **Verification:** without a restriction, 6 of 14 radiosonde dewpoint groups would sit at
  100 hPa or above.

### Fix

- Reject flag 9 for radiosonde `dewPointTemperature`.
- List flag 9 for radiosonde `airTemperature` as well, so the policy is explicit and
  consistent. This is a deliberate no-op: an audit of a 2,000,000-row sample found
  `airTemperatureQuality` is flag 2 in 99.1–99.4% of rows at every band from 1100 hPa to
  below 30 hPa, and flag 9 never occurs.
- Extend `qc_relations` (`dewpoint_le_temp`, `max_temp_dewpoint_spread`) to radiosonde. It
  previously existed only for `surface_obs`, so no Td ≤ T check was applied to radiosonde
  profiles anywhere. The implementation is already generic over each instrument's own
  features, so no code change was needed. `rh_from_td_consistency_pct` is deliberately
  omitted — radiosonde has no `relativeHumidity` feature, so it would silently do nothing.

Both `observation_config.yaml` and `observation_config_finetune.yaml` carry their own copy
of the radiosonde block, so both are changed; otherwise fine-tuning would reintroduce the
unchecked humidity.

### Notes

- The QC runs **at training time**, not in a separate preprocessing pass: the datamodule
  calls `extract_features()` (`process_timeseries.py:559`) on the raw zarr, and the reject
  list is applied to both inputs and every target window (`process_timeseries.py:706, 728`).
  **No data reprocessing is required.**
- The summary cache key includes the full `observation_config`, so it self-invalidates.
  Verified: pre-fix key `49080edc…`, post-fix key `53aaf4e9…`. **No cache clearing required.**

---

### What happens

Satellite rows carry `input_channel_mask`, marking which channels are real. The encoder
never read it, and preprocessing imputes missing channels as normalized zero
(`process_timeseries.py:1178`):

```python
np.nan_to_num(input_features_norm, nan=0.0)
```

Satellite features are standardized per channel, so **normalized zero is the climatological
mean of that channel**, not a neutral placeholder. Using the AVHRR ch_50 statistics from
`configs/observation_config.yaml` (mean 284.58 K, std 17.52 K):

| Row | raw | → model input | `input_channel_mask` |
|---|---|---|---|
| genuinely measured 284.58 K | 284.58 | **0.0000** | 1 |
| **missing (no reading)** | nan | **0.0000** | 0 |
| genuinely measured 310.00 K | 310.00 | 1.4509 | 1 |
| genuinely measured 250.00 K | 250.00 | −1.9737 | 1 |

Rows 1 and 2 reach the encoder as the identical number. A missing channel therefore entered
the model as a perfectly plausible warm-scene observation.

Conventional instruments were never affected: the same function writes a `-9.0` sentinel for
missing values and clips real ones to ±6 (`process_timeseries.py:1270-1272`), so `-9.0` is
unreachable by genuine data and the model can learn it means "absent".

### Impact

- **Training:** every satellite data gap is presented as a real, average-valued observation.
- **Input-ablation experiments:** masking a channel writes that same missing-value
  representation, so `sample_mask` and `full_mask` were substituting a climatological-mean
  observation rather than removing the channel's information pathway. They were not denial
  experiments. (This is why `drop_nodes` is the mode currently recommended for removing an
  instrument. No published result depends on the mask modes.)

### Fix

The encoder can now take one validity column per satellite channel, appended to the input
features in the same way as the existing pressure-level embedding:

```
before:   [ 0.0 ]         → "this channel read 284.58 K"        (wrong for a gap)
after:    [ 0.0 , 1 ]     → "measured, and it read 284.58 K"
          [ 0.0 , 0 ]     → "not measured"
```

Conventional inputs are excluded by design — the `-9.0` sentinel already makes missingness
representable.

Implementation: `gnn_model.py:137` (constructor arg), `:483` (embedder widening),
`:1013` (`_append_channel_validity`), `:1085, :1089` (forward-pass call sites).

### Backward compatibility

This changes the architecture: it widens the observation embedders, so a checkpoint trained
with it off cannot be loaded with it on.

| Default | Governs | Value |
|---|---|---|
| `train_gnn.py` CLI | new training runs | **on** |
| `gnn_model.py` constructor | anything loading an existing checkpoint | **off** |

The split is deliberate. FSOI and the other analysis entry points use
`GNNLightning.load_from_checkpoint(...)` with no overrides; the v1.0 checkpoint predates this
flag, so it has no such hyperparameter and falls back to the constructor default, rebuilding
the v1.0 architecture. `save_hyperparameters()` records the setting in every new checkpoint,
so checkpoints stay self-describing.

Pass `--no-channel_validity_features` to reproduce the v1.0 architecture — required when
resuming or fine-tuning the published checkpoint.

**Known consequence:** `RESUME_FROM_LATEST=1` pointed at a pre-change checkpoint will now
fail with a state-dict shape mismatch unless that flag is passed. Training from scratch is
unaffected.

Once a model is retrained with the feature on, the mask modes become genuine denials,
because `_sync_input_channel_mask_from_values` already clears the stored mask for masked
entries.
